nixos/modules/services/networking/wireguard.nix

```diff
@@ -265,25 +316,8 @@ let
           wg set ${name} private-key ${privKey} ${
             optionalString (values.listenPort != null) " listen-port ${toString values.listenPort}"}
 
-          ${concatMapStringsSep "\n" (peer:
-            assert (peer.presharedKeyFile == null) || (peer.presharedKey == null); # at most one of the two must be set
```

Comment:
This assert has been moved down in to the assertions, which provides a reasonable error message than "assertion failed" :P